pkg/ffmpeg/ffprobe.go

```diff
@@ -10,11 +10,89 @@ import (
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/stashapp/stash/pkg/manager/config"
 )
 
 var ValidCodecs = []string{"h264", "h265", "vp8", "vp9"}
 
+type Container string
+type AudioCodec string
+
+const (
+	Mp4                Container  = "mp4"
+	M4v                Container  = "m4v"
+	Mov                Container  = "mov"
+	Wmv                Container  = "wmv"
+	Webm               Container  = "webm"
+	Matroska           Container  = "matroska"
+	Avi                Container  = "avi"
+	Flv                Container  = "flv"
+	Mpegts             Container  = "mpegts"
+	Aac                AudioCodec = "aac"
+	Mp3                AudioCodec = "mp3"
+	Opus               AudioCodec = "opus"
+	Vorbis             AudioCodec = "vorbis"
+	MissingUnsupported AudioCodec = ""
+)
+
+var validForH264Mkv = []Container{Mp4, Matroska}
+var validForH264 = []Container{Mp4}
+var validForH265Mkv = []Container{Mp4, Matroska}
+var validForH265 = []Container{Mp4}
+var validForVp8 = []Container{Webm}
+var validForVp9Mkv = []Container{Webm, Matroska}
+var validForVp9 = []Container{Webm}
+var validForHevcMkv = []Container{Mp4, Matroska}
+var validForHevc = []Container{Mp4}
+
+var validAudioForMkv = []AudioCodec{Aac, Mp3, Vorbis, Opus}
+var validAudioForWebm = []AudioCodec{Vorbis, Opus}
+var validAudioForMp4 = []AudioCodec{Aac, Mp3}
+
+//maps user readable container strings to ffprobe's format_name
+//on some formats ffprobe can't differentiate
+var ContainerToFfprobe = map[Container]string{
+	Mp4:      "mov,mp4,m4a,3gp,3g2,mj2",
+	M4v:      "mov,mp4,m4a,3gp,3g2,mj2",
+	Mov:      "mov,mp4,m4a,3gp,3g2,mj2",
+	Wmv:      "asf",
+	Webm:     "matroska,webm",
+	Matroska: "matroska,webm",
+	Avi:      "avi",
+	Flv:      "flv",
+	Mpegts:   "mpegts",
+}
+
+var FfprobeToContainer = map[string]Container{
+	"mov,mp4,m4a,3gp,3g2,mj2": Mp4, // browsers support all of them so we don't  care
+	"asf":                     Wmv,
+	"avi":                     Avi,
+	"flv":                     Flv,
+	"mpegts":                  Mpegts,
+	"matroska,webm":           Matroska,
+}
```

Comment:
We're reusing string literals here, so we should create and/or refer to constants for them.